src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,146 @@
+package duke.parser;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+
+/**
+ * The Parser class deals with interpreting the user inputs
+ * and generating corresponding commands if valid.
+ */
+public class Parser {
+    /**
+     * Returns a command based on the user input.
+     * If the command is invalid, an exception is thrown.
+     *
+     * @param userInput The input provided by the user.
+     * @return A Command object representing the user's intended action.
+     * @throws DukeException If there is an error in the user input.
+     */
+    public static Command parse(String userInput) throws DukeException {
+        String[] splitCommand = userInput.trim().split(" ", 2);
+
+        switch (splitCommand[0]) {
+        case "list":
+            return new ListCommand();
+        case "bye":
+            return new ExitCommand();
+        case "event":
+            // Add event task into task list
+            if (!userInput.matches("event .*/from .* /to .*")) {
+                throw new DukeException("OOPS!!! The format of an event task is "
+                        + "\"event TASK_DESCRIPTION /from START /to END\"");
+            }
+
+            String description = splitCommand[1].split("/from")[0];
+            String[] dateAndTime = splitCommand[1].split("/from")[1].split("/to");
+            return new AddCommand(new Event(description, dateAndTime[0], dateAndTime[1]));
+        case "deadline":
+            // Add deadline task into task list
+            return parseDeadlineCommand(splitCommand[1]);
+        case "todo":
+            // Add to-do task into task list
+            if (!userInput.matches("todo .*")) {
+                throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
+            }
+            return new AddCommand(new ToDo(splitCommand[1]));
+        case "mark":
+        case "unmark":
```

Comment:
Maybe u can delete these two lines?

Reply:
Hello, I don't think I should delete these 2 lines. They are not empty cases, but rather have the same method implementation as the "delete" case, hence they are placed together like this. Hope that clarifies!